### PR TITLE
Fix "Boolean" parameters & add csi.storage.k8s.io/fstype

### PR DESCRIPTION
**Pull request by coulof**
# Description
The StorageClass parameters need `String` not `Boolean` ; without quotes it is failing
Add `csi.storage.k8s.io/fstype` examples

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/188 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
All parameters (true/false and different `fsType`) have been tested successfully on a local setup

### CODE REVIEW

File: samples/storageclass/unity-plain.yaml [hunk 1]
```diff
@@ -26,19 +26,25 @@ parameters:
...
   # Optional: false
   # Examples: "Autotyre"
-  hostIOLimitName: <HOST_IO_LIMIT_NAME> 
+  hostIOLimitName: <HOST_IO_LIMIT_NAME>
+  # csi.storage.k8s.io/fstype - Set the filesystem type to format the new volume
+  # Default value: ext4
+  # Accepted values:
+  #   "ext4"
+  #   "xfs"
+  csi.storage.k8s.io/fstype: ext4
```
Comment (AronAtDell) [inline samples/storageclass/unity-plain.yaml:50]: Newline alerts.

Comment (coulof) [reply samples/storageclass/unity-plain.yaml]: Fixed

File: samples/storageclass/unity-fc.yaml [hunk 2]
```diff
@@ -88,12 +88,18 @@ parameters:
...
   # Optional: true
   # Default value: None
   # Examples: "Autotier"
   hostIOLimitName: <HOST_IO_LIMIT_NAME>
+  # csi.storage.k8s.io/fstype - Set the filesystem type to format the new volume
+  # Default value: ext4
+  # Accepted values:
+  #   "ext4"
+  #   "xfs"
+  csi.storage.k8s.io/fstype: xfs
```
Comment (AronAtDell) [inline samples/storageclass/unity-fc.yaml:102]: Was this intentionally meant to be uncommented? Or did we just was to add "... : xfs" to the example while still remaining commented?

File: samples/storageclass/unity-iscsi.yaml [hunk 3]
```diff
@@ -87,13 +87,19 @@ parameters:
...
   # Optional: true
   # Examples: "Autotier"
-  hostIOLimitName: <HOST_IO_LIMIT_NAME> 
+  hostIOLimitName: <HOST_IO_LIMIT_NAME>
+  # csi.storage.k8s.io/fstype - Set the filesystem type to format the new volume
+  # Default value: ext4
+  # Accepted values:
+  #   "ext4"
+  #   "xfs"
+  csi.storage.k8s.io/fstype: xfs
```
Comment (AronAtDell) [inline samples/storageclass/unity-iscsi.yaml:102]: Same as comment above.

File: samples/storageclass/unity-fc.yaml (hunk 2 above)
Comment (coulof) [reply samples/storageclass/unity-fc.yaml:102]: I preferred to make the fstype explicit as we had an issue with NFS provisioner and with non-specified value that was ext4.

Maybe I can comment it out for block and let it be uncommented for NFS.

Let me know if that works for you.

Comment (prablr79) [reply samples/storageclass/unity-fc.yaml:102]: its customer choice, its a sample file, from engineering even XFS works .

**Review by rajendraindukuri — DISMISSED** [review]

**Review by karthikk92 — DISMISSED** [review]

**Review by karthikk92 — APPROVED** [review]

**Review by rensyct — APPROVED** [review]

**Pull request merged**